next/13/replace-api-routes/index.ts

```diff
@@ -0,0 +1,329 @@
+	}
+
+	return newDirectoryNameSegments.join(posix.sep);
+};
+
+const findAPIRouteHandler = (sourceFile: SourceFile): Handler | null => {
+	const defaultExportedFunctionDeclaration = sourceFile
+		.getFunctions()
+		.find((f) => f.isDefaultExport());
+
+	if (defaultExportedFunctionDeclaration !== undefined) {
+		return defaultExportedFunctionDeclaration;
+	}
+
+	const exportAssignment = sourceFile
+		.getStatements()
+		.find((s) => Node.isExportAssignment(s));
+
+	const declarations =
+		exportAssignment
+			?.getFirstDescendantByKind(SyntaxKind.Identifier)
+			?.getSymbol()
+			?.getDeclarations() ?? [];
+
+	let handler:
+		| ArrowFunction
+		| FunctionExpression
+		| FunctionDeclaration
+		| undefined;
+
+	declarations.forEach((d) => {
+		if (Node.isVariableDeclaration(d)) {
+			const initializer = d?.getInitializer();
+
+			if (
+				Node.isArrowFunction(initializer) ||
+				Node.isFunctionExpression(initializer)
+			) {
+				handler = initializer;
+				return;
+			}
+		}
+
+		if (Node.isFunctionDeclaration(d)) {
+			handler = d;
+		}
+	});
+
+	return handler ?? null;
+};
+
+const getPositionAfterImports = (sourceFile: SourceFile): number => {
+	const lastImportDeclaration =
+		sourceFile.getLastChildByKind(SyntaxKind.ImportDeclaration) ?? null;
+
+	return (lastImportDeclaration?.getChildIndex() ?? 0) + 1;
+};
+
+const HTTP_METHODS = ['GET', 'POST', 'PUT', 'DELETE', 'PATCH'] as const;
+
+export type HTTPMethod = (typeof HTTP_METHODS)[number];
+
+const RESPONSE_INIT_FIELDS = ['headers', 'status', 'statusText'] as const;
+
+type ResponseInitParam = (typeof RESPONSE_INIT_FIELDS)[number];
+type ResponseInit = Partial<{ [k in ResponseInitParam]: unknown }>;
+
+const unquotify = (input: string): string => {
+	if (
+		(input.startsWith('"') && input.endsWith('"')) ||
+		(input.startsWith(`'`) && input.endsWith(`'`))
+	) {
+		return input.slice(1, -1);
+	}
+
+	return input;
+};
+
+// res.status() => status
+const getCallExpressionName = (callExpression: CallExpression) => {
+	const expression = callExpression.getExpression();
+
+	if (!Node.isPropertyAccessExpression(expression)) {
+		return null;
+	}
+
+	return expression.getName();
+};
+
+const rewriteResponseCallExpressions = (handler: Handler) => {
+	const responseInit: ResponseInit = {};
+
+	const callExpressions = handler
+		.getDescendantsOfKind(SyntaxKind.CallExpression)
+		.filter((callExpression) => {
+			return getCallExpressionName(callExpression) === 'json';
+		});
+
+	callExpressions.forEach((callExpression) => {
+		const childCallExpressions = callExpression.getDescendantsOfKind(
+			SyntaxKind.CallExpression,
+		);
+
+		childCallExpressions.forEach((childCallExpression) => {
+			const name = getCallExpressionName(childCallExpression);
+
+			if (RESPONSE_INIT_FIELDS.includes(name as ResponseInitParam)) {
+				responseInit[name as ResponseInitParam] = unquotify(
+					childCallExpression.getArguments()[0]?.getText() ?? '',
+				);
+			}
+		});
+
+		const callExpressionArg =
```

Comment:
you can move this above line 139 and use in line 144

Reply:
line 144 is inside loop of childCallExpressions  and `callExpressionArg` is a parent call expression arg. so i think it should be here